src/Types.hs

```diff
@@ -625,3 +629,10 @@ ideaMaybeTopicId = ideaLocation . ideaLocationMaybeTopicId
 isWild :: IdeaLocation -> Bool
 isWild (IdeaLocationSpace _)   = True
 isWild (IdeaLocationTopic _ _) = False
+
+topicToIdeaLocation :: Topic -> IdeaLocation
+topicToIdeaLocation = IdeaLocationTopic <$> (^. topicIdeaSpace) <*> (^. _Id)
+
+userToIdeaLocation :: User -> IdeaLocation
+userToIdeaLocation u =
+    (\(Student clss) -> IdeaLocationSpace (ClassSpace clss)) . head $ u ^. userGroups
```

Comment:
ooooh, `head`.  very bad :-).  but since we have #266, we don't have to do anything about it now.